src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/InputInProcess.cs

```diff
@@ -2,117 +2,124 @@
 
-            public TestServices TestServices { get; }
+            callback(new InputSimulator());
 
-            protected override void ActivateMainWindow()
+            TestServices.JoinableTaskFactory.Run(async () =>
             {
-                TestServices.JoinableTaskFactory.Run(async () =>
-                {
-                    await TestServices.Editor.ActivateAsync(CancellationToken.None);
-                });
-            }
+                await WaitForApplicationIdleAsync(CancellationToken.None);
+            });
+        }
 
-            protected override void WaitForApplicationIdle(CancellationToken cancellationToken)
-            {
-                TestServices.JoinableTaskFactory.Run(async () =>
+        internal Task SendToNavigateToAsync(params InputKey[] keys)
+        {
+            return SendToNavigateToAsync(
+                simulator =>
                 {
-                    await WaitForApplicationIdleAsync(cancellationToken);
+                    foreach (var key in keys)
+                    {
+                        key.Apply(simulator);
+                    }
                 });
-            }
         }
 
-        private class SendKeysWithoutActivateImpl : SendKeysImpl
+        internal async Task SendToNavigateToAsync(Action<IInputSimulator> callback)
         {
-            public SendKeysWithoutActivateImpl(TestServices testServices)
-                : base(testServices)
+            // AbstractSendKeys runs synchronously, so switch to a background thread before the call
+            await TaskScheduler.Default;
+
+            // Take no direct action regarding activation, but assert the correct item already has focus
+            TestServices.JoinableTaskFactory.Run(async () =>
             {
-            }
+                await TestServices.JoinableTaskFactory.SwitchToMainThreadAsync();
+                var searchBox = Assert.IsAssignableFrom<TextBox>(Keyboard.FocusedElement);
+                Assert.Equal("PART_SearchBox", searchBox.Name);
+            });
 
-            protected override void ActivateMainWindow()
+            callback(new InputSimulator());
+
+            TestServices.JoinableTaskFactory.Run(async () =>
             {
-                // Take no direct action
-            }
+                await WaitForApplicationIdleAsync(CancellationToken.None);
+            });
         }
 
-        private class SendKeysToNavigateToImpl : SendKeysImpl
+        internal async Task MoveMouseAsync(Point point)
         {
-            public SendKeysToNavigateToImpl(TestServices testServices)
-                : base(testServices)
-            {
-            }
+            var horizontalResolution = NativeMethods.GetSystemMetrics(NativeMethods.SM_CXSCREEN);
+            var verticalResolution = NativeMethods.GetSystemMetrics(NativeMethods.SM_CYSCREEN);
+            var virtualPoint = new ScaleTransform(65535.0 / horizontalResolution, 65535.0 / verticalResolution).Transform(point);
```

Comment:
@sharwell  could you give a brief explanation of these magic numbers?

Reply:
See here:
https://github.com/TChatzigiannakis/InputSimulatorPlus/blob/af96f2a453d48f632a5c98153a6aa87fb1a26012/WindowsInput/MouseSimulator.cs#L71-L105

The current implementation won't work in multi-monitor configurations where the test VS instance is not on the primary display device. This isn't something we have to worry about in CI but hopefully we can fix it later by switching from `MoveMouseTo` to `MoveMouseToPositionOnVirtualDesktop`.